ods_ci/tests/Tests/1300__model_registry/1302_model_registry_model_serving.robot

```diff
@@ -0,0 +1,277 @@
+    Sleep    2m    msg=Waits until the jupyter notebook has completed execution of all cells
+    JupyterLab Code Cell Error Output Should Not Be Visible
+    SeleniumLibrary.Capture Page Screenshot
+
+Wait For Model Registry Containers To Be Ready
+    [Documentation]    Wait for model-registry-deployment to be ready
+    ${result}=    Run Process
+    ...        oc wait --for\=condition\=Available --timeout\=5m -n ${PRJ_TITLE} deployment/model-registry-db
+    ...        shell=true    stderr=STDOUT
+    Log To Console    ${result.stdout}
+    ${result}=    Run Process
+    ...        oc wait --for\=condition\=Available --timeout\=5m -n ${PRJ_TITLE} deployment/model-registry-deployment
+    ...        shell=true    stderr=STDOUT
+    Log To Console    ${result.stdout}
+
+Get Domain
+    [Documentation]  Gets the Domain and returns it to 'Run Oc Commands And Capture Output'.
+    # Run the command to get the ingress domain
+    ${domain_result}=    Run Process    oc    get    ingresses.config/cluster
+    ...    -o    yaml    stdout=PIPE    stderr=PIPE
+    ${rc}=    Set Variable    ${domain_result.rc}
+    IF    $rc > 0    Fail    Command 'oc whoami -t' returned non-zero exit code: ${rc}
+    ${domain_yaml_output}=    Set Variable    ${domain_result.stdout}
+
+    # Return the domain from stdout
+    ${domain_parsed_yaml}=    Evaluate    yaml.load('''${domain_yaml_output}''', Loader=yaml.FullLoader)
+    ${ingress_domain}=    Set Variable    ${domain_parsed_yaml['spec']['domain']}
+
+    # Return both results
+    RETURN    ${ingress_domain}
+
+Get Token
+    [Documentation]    Gets the Token and returns it to 'Run Oc Commands And Capture Output'.
+    ${token_result}=    Run Process    oc    whoami    -t    stdout=YES
+    ${rc}=    Set Variable    ${token_result.rc}
+    IF    ${rc} > 0    Fail    Command 'oc whoami -t' returned non-zero exit code: ${rc}
+    ${token}=    Set Variable    ${token_result.stdout}
+    RETURN    ${token}
+
+Create OpenShift Project
+    [Documentation]    Creates a test namespace and tracks it via ServiceMesh
+    [Arguments]    ${NAMESPACE_MODEL-REGISTRY}
+    ${rc}    ${out}=    Run And Return Rc And Output    oc get project ${NAMESPACE_MODEL-REGISTRY}
+    Log    ${out}
+    IF    ${rc} == ${0}
+        Log    message=OpenShift Project ${NAMESPACE_MODEL-REGISTRY} already present. Skipping project setup...
+        ...    level=WARN
+        RETURN
+    END
+    ${rc}    ${output}=    Run And Return Rc And Output    oc new-project ${NAMESPACE_MODEL-REGISTRY}
+    Log    ${output}
+    Should Be Equal As Numbers    ${rc}    ${0}
+
+Apply ServiceMeshMember Configuration
+    [Documentation]    Apply a ServiceMeshMember configuration using oc.
+    Apply OpenShift Configuration    ${SERVICE_MESH_MEMBER}
+
+Apply Rest API Configuration
+    [Documentation]    Apply a Rest API configuration using oc.
+    Apply OpenShift Configuration    ${ENABLE_REST_API}
+
+Apply OpenShift Configuration
```

Comment:
maybe this keyword could be moved in a common resource file to be re-used by others, for example `ods_ci/tests/Resources/OCP.resource`

Reply:
This seems to only be calling `oc apply`, I'm not really sure we need a keyword for it?
And to be honest we probably already have some keyword(s) that just apply a file, so in that case it might make sense to centralize all of them under one keyword?

Reply:
I agree with both of you. If something like that exists then I should use it, if not it should go into a common file for others to use. I'll check it out.

Reply:
I have put the keyword into OCP.resource and called it from there.  I searched for another function but couldn't see any.  I think it makes sense to have a keyword instead of directly calling oc apply -f for each time it is needed. It has fewer lines of code with the keyword.